Source/Dafny/Verifier/Translator.ClassMembers.cs

```diff
@@ -1382,9 +1383,12 @@ public partial class Translator {
         foreach (var formal in m.Ins) {
           if (formal.IsOld) {
             var dafnyFormalIdExpr = new IdentifierExpr(formal.tok, formal);
-            req.Add(Requires(formal.tok, false, MkIsAlloc(etran.TrExpr(dafnyFormalIdExpr), formal.Type, prevHeap),
-              string.Format("parameter{0} ('{1}') must be allocated in the two-state lemma's previous state",
-                m.Ins.Count == 1 ? "" : " " + index, formal.Name), null));
+            var pIdx = m.Ins.Count == 1 ? "" : " " + index;
```

Comment:
Let me know if this change cause problem

Reply:
I'm undecided about this. I tend to use the phrase "parameter 1" in practice more than the phrase "parameter at index 1".

Reply:
I would say that "parameter 2" is the equivalent of "parameter at index 1", not "parameter 1", simply because indexing in everyday life consists of assigning the number of the elements already seen to the new element, that new element included.
i would wholeheartedly prefer "parameter 2" instead of "parameter at index 1", but for consistency with other error messages, until we decide to change "at index" to a more human-readable indexing numbering system, I would really prefer to warn users that "parameter 1" really means the second parameter.